### ISSUE TITLE
Possibility of reading too many bytes when reading NBT strings

### ISSUE THREAD

**Issue by UnknownShadow200** [fragmer/fNbt#26]
In NbtBinaryReader.cs:
```CSharp
while (stringBytesRead < length) {
  int bytesReadThisTime = BaseStream.Read(stringConversionBuffer, stringBytesRead, length);
  if (bytesReadThisTime == 0) {
    throw new EndOfStreamException();
  }
  stringBytesRead += bytesReadThisTime;
}
```
Should be ```length -  stringBytesRead``` instead of just ```length```

**Comment by fragmer** [comment]
This has been fixed in `master`, thanks! I'll close this when I bake a fresh NuGet package for 0.6.4, later today.

**Comment by fragmer** [comment]
Fix released with [v0.6.4](https://github.com/fragmer/fNbt/releases/tag/v0.6.4)
